### PR TITLE
refactor: `main` functions.

### PR DESCRIPTION
**Pull request by ninaandrup**
Related to https://github.com/flix/flix/issues/3518

### CODE REVIEW

File: src/content/TipsAndTricks.js [hunk 1]
```diff
@@ -30,21 +30,22 @@ class TipsAndTricks extends React.Component {
...
                     <CodeBlock>
-                        {`def main(args: Array[String]): Int32 & Impure = ...`}
+                        {`def main(): Unit & Impure = ...`}
                     </CodeBlock>
 
                     <p>
-                        That is, the main function (i) takes an array of string arguments, (ii) must return
-                        an <Code>Int32</Code> which represents the exit code, and (iii) must be <Code>Impure</Code>.
+                        That is, the main function (i) must return <Code>Unit</Code>, (ii) must be <Code>Impure</Code>,
+                        and (iii) must be private.
```
Comment (ninaandrup) [inline src/content/TipsAndTricks.js]: Are these the correct requirements for the new `main`?

File: src/content/TipsAndTricks.js [hunk 2]
```diff
@@ -30,21 +30,22 @@ class TipsAndTricks extends React.Component {
...
                     </p>
 
                     <p>
-                        Flix does not allow unused function arguments. Thus, if the arguments are not needed the
-                        parameter name must be prefixed with an underscore:
+                        If <Code>main</Code> is run with arguments, they can be accessed by writing:
```
Comment (ninaandrup) [inline src/content/TipsAndTricks.js]: I'm not sure, if I like this sentence...

File: src/content/TipsAndTricks.js (hunk 1 above)
Comment (magnus-madsen) [reply src/content/TipsAndTricks.js]: It does not have to be private. (We changed all top level to be implicitly public). So if you keep (i) and (ii) its good 👍

File: src/content/TipsAndTricks.js (hunk 2 above)
Comment (magnus-madsen) [reply src/content/TipsAndTricks.js]: Maybe: The command line arguments passed to the program can be accessed by calling `Environment.getArgs()` or something to that effect :)

File: src/content/TipsAndTricks.js [hunk 3]
```diff
@@ -30,21 +30,23 @@ class TipsAndTricks extends React.Component {
...
-                        an <Code>Int32</Code> which represents the exit code, and (iii) must be <Code>Impure</Code>.
+                        That is, the main function (i) must return <Code>Unit</Code>, and (ii) must
+                        be <Code>Impure</Code>.
                     </p>
 
                     <p>
-                        Flix does not allow unused function arguments. Thus, if the arguments are not needed the
-                        parameter name must be prefixed with an underscore:
+                        The signature of <Code>main</Code> does not specify any arguments, but the command line
+                        arguments passed to the program can be accessed by calling <Code>Environment.getArgs()</Code>.
```
Comment (ninaandrup) [inline src/content/TipsAndTricks.js:43]: Is this okay?

Comment (magnus-madsen) [reply src/content/TipsAndTricks.js:43]: 👍

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**